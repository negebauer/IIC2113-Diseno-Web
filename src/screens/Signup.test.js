import React from 'react'
import createHistory from 'history/createBrowserHistory'
import { Provider } from 'react-redux'
import { ConnectedRouter } from 'react-router-redux'
import { shallow, mount } from 'enzyme'
import toJson from 'enzyme-to-json'
import ConnectedSignup, { Signup } from './Signup'
import configureStore from '../redux/store'

let history
let initialState
let store

beforeEach(() => {
  history = createHistory()
  initialState = {}
  const api = { signup: () => {}, saveUser: () => {} }
  store = configureStore(initialState, { history, api })
})

it('shallow renders without crashing', () => {
  shallow(<ConnectedSignup store={store} />)
})

it('mount renders without crashing', () => {
  mount(
    <Provider store={store}>
      <ConnectedRouter history={history}>
        <ConnectedSignup />
      </ConnectedRouter>
    </Provider>
  )
})

it('matches snapshot', () => {
  const data = { loading: false, api_key: '', error: '' }
  const wrapper = mount(
    <Signup
      {...data}
      signup={() => {}}
      saveUser={() => {}}
      replace={() => {}}
    />
  )
  expect(toJson(wrapper)).toMatchSnapshot()
})

it('doesnt render children', () => {
  const wrapper = mount(
    <Provider store={store}>
      <ConnectedRouter history={history}>
        <ConnectedSignup />
      </ConnectedRouter>
    </Provider>
  )
  expect(wrapper.children().exists()).toBeTruthy()
})
