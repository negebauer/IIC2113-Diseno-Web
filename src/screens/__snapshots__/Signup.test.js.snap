// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<Signup
  api_key=""
  error=""
  loading={false}
  replace={[Function]}
  saveUser={[Function]}
  signup={[Function]}
>
  <div
    className="container"
  >
    <div
      className="row"
      id="signup"
    >
      <form
        className="col s12"
      >
        <h2
          className="header"
        >
          Signup
        </h2>
        <div
          className="row"
        >
          <div
            className="input-field col s12"
          >
            <input
              className="validate"
              id="first_name"
              name="name"
              onChange={[Function]}
              type="text"
              value=""
            />
            <label
              htmlFor="name"
            >
              Nombre
            </label>
          </div>
        </div>
        <div
          className="row"
        >
          <div
            className="input-field col s12"
          >
            <input
              className="validate"
              id="mail"
              name="mail"
              onChange={[Function]}
              type="email"
              value=""
            />
            <label
              htmlFor="email"
            >
              Email
            </label>
          </div>
        </div>
        <div
          className="row"
        >
          <div
            className="input-field col s12"
          >
            <input
              className="validate"
              id="password"
              name="password"
              onChange={[Function]}
              type="password"
              value=""
            />
            <label
              htmlFor="password"
            >
              Password
            </label>
          </div>
        </div>
        <div
          className="row"
        >
          <div
            className="input-field col s12"
          >
            <input
              className="validate"
              id="password_confirmation"
              name="password_confirmation"
              onChange={[Function]}
              type="password"
              value=""
            />
            <label
              htmlFor="password"
            >
              Password Confirmation
            </label>
          </div>
        </div>
        <button
          className="waves-effect waves-light btn-large"
          onClick={[Function]}
        >
          Signup
        </button>
      </form>
    </div>
  </div>
</Signup>
`;
